app/screens/notification-history-screen/notification.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  StatefulNotification,
+  StatefulNotificationsDocument,
+  useStatefulNotificationAcknowledgeMutation,
+} from "@app/graphql/generated"
+import { Text, makeStyles } from "@rneui/themed"
+import { View } from "react-native"
+import { timeAgo } from "./utils"
+import { gql } from "@apollo/client"
+import { TouchableWithoutFeedback } from "react-native-gesture-handler"
+import { useLinkTo } from "@react-navigation/native"
+import { useState } from "react"
+
+gql`
+  mutation StatefulNotificationAcknowledge(
+    $input: StatefulNotificationAcknowledgeInput!
+  ) {
+    statefulNotificationAcknowledge(input: $input) {
+      notification {
+        acknowledgedAt
+      }
+    }
+  }
+`
+
+export const Notification: React.FC<StatefulNotification> = ({
+  id,
+  title,
+  body,
+  createdAt,
+  acknowledgedAt,
+  deepLink,
+}) => {
+  const [isAcknowledged, setIsAcknowledged] = useState(Boolean(acknowledgedAt))
+  const styles = useStyles({ isAcknowledged })
+
+  const [ack, _] = useStatefulNotificationAcknowledgeMutation({
+    variables: { input: { notificationId: id } },
+    refetchQueries: [StatefulNotificationsDocument],
+  })
+
+  const linkTo = useLinkTo()
+
+  return (
+    <TouchableWithoutFeedback
+      onPress={() => {
+        setIsAcknowledged(true)
+        !isAcknowledged && ack()
+        deepLink && linkTo(deepLink)
```

Comment:
Do you think we should have a separate button for following the link. It could be confusing if you are just trying to mark a notification as read and all of a sudden you are linked to a different page.

Reply:
I was thinking about this too but most other apps "react" on tap of notifications and they don't have a different button for that. Confused about this UX.